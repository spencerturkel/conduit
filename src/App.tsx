import * as React from 'react';
import {ComponentClass} from 'react';
import {BrowserRouter, Link, NavLink} from 'react-router-dom';

import Footer, {LinkProps as FooterLinkProps} from './Footer/Footer';
import Header, {LinkProps as HeaderLinkProps} from './Header/Header';
import Home from './Home/Home';

const AppHeader = Header(NavLink as ComponentClass<HeaderLinkProps>);
const AppFooter = Footer(Link as ComponentClass<FooterLinkProps>);

const App = () => (
    <BrowserRouter>
        <div>
            <AppHeader />
            <Home />
            <AppFooter />
        </div>
    </BrowserRouter>
);

export default App;
