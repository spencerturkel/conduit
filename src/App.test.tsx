import {shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import * as ReactDOM from 'react-dom';

import App from './App';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

describe('shallow tests', () => {
    let sut: ShallowWrapper;

    beforeEach(() => {
        sut = shallow(<App />);
    });

    it('matches the snapshot', () => {
        expect(sut).toMatchSnapshot();
    });
});
