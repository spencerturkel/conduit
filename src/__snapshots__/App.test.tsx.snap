// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow tests matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Header(NavLink) />
        <Home(AppPopularTags, ArticleListing(FeedPicker))
                previews={
                        Array [
                                Object {
                                  "author": Object {
                                    "image": "https://i.stack.imgur.com/xHWG8.jpg",
                                    "username": "jake",
                                  },
                                  "createdAt": 2017-12-10T00:00:00.000Z,
                                  "description": "This is the description for the post.",
                                  "favoritesCount": 27,
                                  "slug": "test-slug",
                                  "title": "How to build webapps that scale",
                                },
                              ]
                }
        />
        <Footer(Link) />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header(NavLink) />,
          <Home(AppPopularTags, ArticleListing(FeedPicker))
            previews={
                        Array [
                                    Object {
                                      "author": Object {
                                        "image": "https://i.stack.imgur.com/xHWG8.jpg",
                                        "username": "jake",
                                      },
                                      "createdAt": 2017-12-10T00:00:00.000Z,
                                      "description": "This is the description for the post.",
                                      "favoritesCount": 27,
                                      "slug": "test-slug",
                                      "title": "How to build webapps that scale",
                                    },
                                  ]
            }
/>,
          <Footer(Link) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "previews": Array [
              Object {
                "author": Object {
                  "image": "https://i.stack.imgur.com/xHWG8.jpg",
                  "username": "jake",
                },
                "createdAt": 2017-12-10T00:00:00.000Z,
                "description": "This is the description for the post.",
                "favoritesCount": 27,
                "slug": "test-slug",
                "title": "How to build webapps that scale",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Header(NavLink) />
          <Home(AppPopularTags, ArticleListing(FeedPicker))
                    previews={
                              Array [
                                        Object {
                                          "author": Object {
                                            "image": "https://i.stack.imgur.com/xHWG8.jpg",
                                            "username": "jake",
                                          },
                                          "createdAt": 2017-12-10T00:00:00.000Z,
                                          "description": "This is the description for the post.",
                                          "favoritesCount": 27,
                                          "slug": "test-slug",
                                          "title": "How to build webapps that scale",
                                        },
                                      ]
                    }
          />
          <Footer(Link) />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header(NavLink) />,
            <Home(AppPopularTags, ArticleListing(FeedPicker))
              previews={
                            Array [
                                          Object {
                                            "author": Object {
                                              "image": "https://i.stack.imgur.com/xHWG8.jpg",
                                              "username": "jake",
                                            },
                                            "createdAt": 2017-12-10T00:00:00.000Z,
                                            "description": "This is the description for the post.",
                                            "favoritesCount": 27,
                                            "slug": "test-slug",
                                            "title": "How to build webapps that scale",
                                          },
                                        ]
              }
/>,
            <Footer(Link) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "previews": Array [
                Object {
                  "author": Object {
                    "image": "https://i.stack.imgur.com/xHWG8.jpg",
                    "username": "jake",
                  },
                  "createdAt": 2017-12-10T00:00:00.000Z,
                  "description": "This is the description for the post.",
                  "favoritesCount": 27,
                  "slug": "test-slug",
                  "title": "How to build webapps that scale",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
